### PR TITLE
 User can register

### PR DESCRIPTION
**Pull request by OskarCA**
#### Description

- Implement devise for registering user
- Link register In navbar to registration path
- Written shit lots of test.

#### What I have learned working on this feature: 

-Acceptance test & Unit tests
-Reading error messages carefully
-Google tools
-More Insight on rails

#### Screenshots:
screen shot 2018-03-02 at 17 39 03

 
screen shot 2018-03-02 at 17 36 25

screen shot 2018-03-02 at 17 37 52

screen shot 2018-03-02 at 17 36 55

screen shot 2018-03-02 at 17 37 23

screen shot 2018-03-02 at 17 38 14

### CODE REVIEW

File: spec/models/user_spec.rb [hunk 1]
```diff
@@ -0,0 +1,13 @@
+require 'rails_helper'
+
+RSpec.describe User, type: :model do
```
Comment (oliverochman) [inline spec/models/user_spec.rb:3]: you guys really need to format this :)

File: app/views/layouts/application.html.erb [hunk 2]
```diff
@@ -8,6 +8,10 @@
     <%= javascript_include_tag 'application', 'data-turbolinks-track': 'reload' %>
   </head>
   <body>
+      <% if flash[:notice] %>
+          <%= flash[:notice] %>
+      <% end %>
+
```
Comment (tochman) [inline app/views/layouts/application.html.erb]: This should UNDER the navbar IMO

File: features/step_definitions/register_steps.rb [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+
+Then("I should see the message {string}") do |content|
+    expect(page).to have_content content
+end
+
+Given("the following user exist") do |table|
+    table.hashes.each do |user|
+        FactoryBot.create(:user, user)
+   end
+end
```
Comment (tochman) [inline features/step_definitions/register_steps.rb]: Code formatting is off in this file. You should use the same standard as the rest of your team. Check with them if they use 2 spaces or 1 tab, and use the same.

**Review by tochman — COMMENTED** [review]
Good job, but I need you to format your files consistently.

**Review by tochman — APPROVED** [review]
can't really tell if the feature file is properly formatted, but I'll give this one a green light now. Good job!

**Pull request merged**